yml/apdb.yaml

```diff
@@ -1,7 +1,7 @@
 ---
 name: "ApdbSchema"
 "@id": "#apdbSchema"
-version: "1.0.0"
+version: "1.1.0"
```

Comment:
Since it seems like ApdbSchema [is using a "fully/partially/non-compatible" approach](https://dmtn-269.lsst.io/#implementation-path) instead of semantic versioning, I don't think this number is correct -- if you want to argue that it's compatible because the fields were unused, then it should be 1.0.1; if you want to consider the schema on it's own merits, it's 2.0.0. I assume the former is closer to your original intent.

Reply:
I can see arguments for either, but @andy-slac had suggested 1.1.0 in a DM so I went with that.

Reply:
I'm a bit worried by the definition of "backward compatibility" implied by that statement.

Normally when people talk about a library (or its API changes) being backward compatible, they mean that old clients can use new versions of the library. That's also what we assumed it meant in Prompt Processing.

However, you seem to be saying that an APDB change is compatible if _new_ clients can interact with _old_ databases, i.e. forward compatibility in the conventional sense. That's surprising because such compatibility is normally the client's responsibility. It would also mean that in PP we need to be stricter about saying that e.g. a given release can work with schemas 1.0-1.1, but can't guarantee 1.2.

If that's the direction you want compatibility to be defined in, please document that very explicitly.

Reply:
Well, maybe term "backward" is not quite right here, but it more or less reflects the same direction of upgrades. With API example you upgrade library and keep old client. With schema version - you upgrade yaml schema file but can keep old database schema. I'm not sure one can say who is the client of what here, but upgrades usually happen for yaml schema first.

Reply:
Still, please make that unambiguous in the docs (in practice, DMTN-269?) so that everybody has the same expectations.